src/main/kotlin/org/rust/lang/core/psi/impl/RustNumericLiteralImpl.kt

```diff
@@ -0,0 +1,14 @@
+package org.rust.lang.core.psi.impl
+
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.RustLiteralTokenType
+import org.rust.lang.core.psi.RustNumericLiteral
+
+class RustNumericLiteralImpl(type: IElementType, text: CharSequence) : RustLiteralImplBase(type, text), RustNumericLiteral {
+    override fun toString(): String = "RustNumericLiteralImpl($tokenType)"
+
+    companion object {
+        @JvmStatic fun createTokenType(debugName: String): RustLiteralTokenType =
+            RustLiteralTokenType(debugName, ::RustNumericLiteralImpl)
```

Comment:
Neat! I didn't know that it is possible to use constructor as a function in Kotlin!